shard/src/ext/lavalink.py

```diff
@@ -0,0 +1,178 @@
+import re
+import discord
+import lavalink
+from discord.ext import commands
+from src.utils import format_seconds
+
+url_rx = re.compile(r'https?://(?:www\.)?.+')
+
+
+class LavaMusic(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.Cog.listener()
+    async def on_ready(self):
+        if not hasattr(self.bot, 'lavalink'):
+            self.bot.lavalink = lavalink.Client(self.bot.user.id, shard_count=self.bot.shard_dict['shard_count'])
+            self.bot.lavalink.add_node('lavalink', 2333, 'merryandpippinarecute', 'us' 'arch music')
+            self.bot.add_listener(self.bot.lavalink.voice_update_handler, 'on_socket_response')
+            lavalink.add_event_hook(self.track_hook)
+
+    def cog_unload(self):
+        self.bot.lavalink._event_hooks.clear()
+
+    async def cog_before_invoke(self, ctx):
```

Comment:
does discord.py just always call this before a command?

Reply:
Yeah. Before any command in this cog is called that method is run.